src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,422 @@
+          <footer
+            class={{
+              [CSS.footer]: true,
+              [CSS.readonly]: this.readonly,
+              [CSS.hide]: !hasFooter
+            }}
+            key={CSS.footer}
+            ref={(el) => (this.footerEl = el as HTMLElement)}
+          >
+            <slot name={SLOTS.footerLeading} onSlotchange={this.footerTrailingSlotChangeHandler} />
+            {this.renderCharacterLimit()}
+            <slot name={SLOTS.footerTrailing} onSlotchange={this.footerLeadingSlotChangeHandler} />
+          </footer>
+        )}
+        <HiddenFormInputSlot component={this} />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Set's focus on the `textarea`. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.textareaEl.focus();
+  }
+
+  /** Selects all text of the component's `value`. */
+  @Method()
+  async selectText(): Promise<void> {
+    this.textareaEl.select();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  formEl: HTMLFormElement;
+
+  defaultValue: Textarea["value"];
+
+  labelEl: HTMLCalciteLabelElement;
+
+  textareaEl: HTMLTextAreaElement;
+
+  footerEl: HTMLElement;
+
+  @State() effectiveLocale = "";
+
+  @State() trailingSlotElements: Element[];
+
+  @State() leadingSlotElements: Element[];
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @State() defaultMessages: TextareaMessages;
+
+  timeOutId: number;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  onFormReset(): void {
+    this.value = this.defaultValue;
+  }
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  handleInput = (event: InputEvent): void => {
+    this.value = event.target["value"];
+    this.calciteTextareaInput.emit();
+  };
+
+  handleChange = (): void => {
+    this.calciteTextareaChange.emit();
+  };
+
+  footerTrailingSlotChangeHandler = (event: Event): void => {
+    this.trailingSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.trailingSlotElements);
+    }
+  };
+
+  footerLeadingSlotChangeHandler = (event: Event): void => {
+    this.leadingSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.leadingSlotElements);
+    }
+  };
+
+  contentSlotChangeHandler = (): void => {
+    if (!this.value) {
+      const nodes = this.el.childNodes;
```

Comment:
Can you just use `this.el.textContent.trim()`?

If not, can you use the slot event's assigned elements instead of looping through the children? I think it should be more accurate.

```
contentSlotChangeHandler = (event: Event): void => {
    this.value = slotChangeGetAssignedElements(event).map((el) => el.textContent).toString() ;
}
```

Reply:
`slotChangeGetAssignedElements()` wouldn't work in this scenario as the method look for event.target as `HTMLSlotElement` which in this case is a plain text rather than an Element.

Reply:
and the Event emitted from slot change doesn't capture textContent without an element.

Reply:
Reverting back to looping through childNodes and filter nodes with name `text` as `this.el.textContent.trim()` will fetch the text from slotted elements too.